swap/src/monero/wallet.rs

```diff
@@ -261,8 +262,70 @@ impl Wallet {
         self.main_address
     }
 
-    pub async fn refresh(&self) -> Result<Refreshed> {
-        Ok(self.inner.lock().await.refresh().await?)
+    pub async fn refresh(&self, max_attempts: usize) -> Result<Refreshed> {
+        const GET_HEIGHT_INTERVAL: Duration = Duration::from_secs(5);
+        const RETRY_INTERVAL: Duration = Duration::from_secs(2);
+
+        let inner = self.inner.lock().await;
```

Comment:
@binarybaron doing a bit of debugging on this, it appears this line never returns, at least for `given_alice_and_bob_manually_refund_after_funds_locked_both_refund`

Reply:
Stack trace looks like 
```
swap::monero::wallet::Wallet::refresh::{{closure}} wallet.rs:269
swap::monero::wallet::Wallet::create_from::{{closure}} wallet.rs:148
swap::protocol::alice::state::State3::refund_xmr::{{closure}} state.rs:527
swap::asb::recovery::refund::refund::{{closure}} refund.rs:82
alice_and_bob_refund_using_cancel_and_refund_command::given_alice_and_bob_manually_refund_after_funds_locked_both_refund::{{closure}}::{{closure}}::{{closure}} alice_and_bob_refund_using_cancel_and_refund_command.rs:76
alice_and_bob_refund_using_cancel_and_refund_command::harness::setup_test::{{closure}} mod.rs:140
alice_and_bob_refund_using_cancel_and_refund_command::given_alice_and_bob_manually_refund_after_funds_locked_both_refund::{{closure}} alice_and_bob_refund_using_cancel_and_refund_command.rs:82
```

Inside `refund_xmr`, immediately before the `create_from` call there is a call to `watch_for_transfer` which itself calls `wait_for_confirmations` which also locks the client mutex. From the little I know about these locks they should unlock when it goes out of scope, but perhaps there is something going on with the borrowing that keeps it in scope? Not sure.

Reply:
Thanks a lot. @delta1 do you know what exact issue could be? I haven't been able to figure it out.

Reply:
I don't even understand why the HTTP Clients needs to be a Mutex. That seems so unnecessary and complicates everything...